service/smithyprototype/lexruntimeservice/serde.go

```diff
@@ -0,0 +1,542 @@
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+
+	return next.HandleSerialize(ctx, in)
+}
+
+type awsRestJson1_serializeOpPostContent struct{}
+
+func (p awsRestJson1_serializeOpPostContent) ID() string {
+	return "awsRestJson1_serializeOpPostContent"
+}
+
+func (p awsRestJson1_serializeOpPostContent) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	request, ok := in.Request.(*smithyHTTP.Request)
+	if !ok {
+		return out, metadata, &aws.SerializationError{Err: fmt.Errorf("unknown transport type %T", in.Request)}
+	}
+
+	params, ok := in.Parameters.(*PostContentInput)
+	if !ok {
+		return out, metadata, &aws.SerializationError{Err: fmt.Errorf("unknown params parameters type %T", in.Parameters)}
+	}
+
+	restEncoder := rest.NewEncoder(request.Request)
+
+	if err := awsRestJson1_serializeRestPostContentInput(params, restEncoder); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+
+	if err := restEncoder.Encode(); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+
+	// TODO: If MediaType trait was applied in the Smithy model the content-type should get reflected as such
+	if request, err = request.SetStream(params.InputStream); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+	in.Request = request
+
+	return next.HandleSerialize(ctx, in)
+}
+
+type awsRestJson1_serializeOpPostText struct{}
+
+func (p awsRestJson1_serializeOpPostText) ID() string {
+	return "awsRestJson1_serializeOpPostText"
+}
+
+func (p awsRestJson1_serializeOpPostText) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	request, ok := in.Request.(*smithyHTTP.Request)
+	if !ok {
+		return out, metadata, &aws.SerializationError{Err: fmt.Errorf("unknown transport type %T", in.Request)}
+	}
+
+	input, ok := in.Parameters.(*PostTextInput)
+	if !ok {
+		return out, metadata, &aws.SerializationError{Err: fmt.Errorf("unknown input parameters type %T", in.Parameters)}
+	}
+
+	restEncoder := rest.NewEncoder(request.Request)
+
+	restEncoder.AddHeader("Content-Type").String("application/json")
+
+	if err := awsRestJson1_serializeRestPostTextInput(input, restEncoder); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+
+	if err := restEncoder.Encode(); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+
+	jsonEncoder := json.NewEncoder()
+	if err := awsRestJson1_serializeJsonPostTextInput(input, jsonEncoder.Value); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+
+	if request, err = request.SetStream(bytes.NewReader(jsonEncoder.Bytes())); err != nil {
+		return out, metadata, &aws.SerializationError{Err: err}
+	}
+	in.Request = request
```

Comment:
Does it make sense for teh `json.NewEncoder` to take a buffer to write to?  That would save the need to create another buffer later when calling `SetStream`

Reply:
Then wouldn't need a `Bytes` method on the encoder? it would write to the underlying buffer, similar to std json.Encoder, and similar utilities.

Reply:
Yeah agreed let's follow that up in a separate PR as that requires changes out side of the scope and intentions here.